Add merge insertion sort

ryuta69 opened the pull request:
### **Describe your change:**



* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

Wikipedia URL

https://en.wikipedia.org/wiki/Merge-insertion_sort

I did black and flake 8 process.

ruppysuppy commented on sorts/merge_insertion_sort.py:
```diff
@@ -11,21 +11,21 @@
 
 
-def mergeInsertionSort(collection):
+def merge_insertion_sort(collection):
```
Add type hints

ryuta69 replied:
@ruppysuppy 
fixed here
23adf96

ryuta69 commented:
@ruppysuppy 
thank you for reviewing. 

> Add the wikipedia url in the file as a comment or in the docstring

fixed here
50c4185

cclauss commented on sorts/merge_insertion_sort.py:
```diff
@@ -0,0 +1,106 @@
+Source: https://en.wikipedia.org/wiki/Merge-insertion_sort
+
+For doctests run following command:
+python -m doctest -v merge_insertion_sort.py
+or
+python3 -m doctest -v merge_insertion_sort.py
```
```suggestion
python3 -m doctest -v merge_insertion_sort.py
or
python -m doctest -v merge_insertion_sort.py
```
Let's do python3 first because python might still point to legacy Python.

cclauss commented on sorts/merge_insertion_sort.py:
```diff
@@ -0,0 +1,106 @@
+python3 -m doctest -v merge_insertion_sort.py
+
+For manual testing run:
+python merge_insertion_sort.py
```
```suggestion
python3 merge_insertion_sort.py
```

cclauss commented on sorts/merge_insertion_sort.py:
```diff
@@ -0,0 +1,106 @@
+            if left == right:
+                if sorted_list[middle] < item:
+                    left = middle + 1
+                    break
+                else:
+                    break
```
```suggestion
                break
```

ryuta69 commented:
@cclauss 
thank you for reviewing.
I fixed your three requests.

ryuta69 commented:
@poyea 

> substituting single or odd or something else for surplus

Fixed here ad92692

> comments are encouraged too

Added here f0d1215. If this comment is too long, please tell me, I will fix.
